politeiawww/proposals_test.go

```diff
@@ -408,40 +408,36 @@ func TestValidateProposal(t *testing.T) {
 	// Duplicate filenames
 	propDupFiles := createNewProposal(t, id, []www.File{*md, *png, *png})
 
-	// Too many markdown files. We need one correctly named md
+	// Attachment is duplicate of index file
+	attachmentFile := createFileMD(t, 8, "Valid Title")
+	propAttachmentIndexDup := createNewProposal(t, id, []www.File{*md, *attachmentFile})
+
+	// Invalid proposal title
+	mdBadTitle := createFileMD(t, 8, "{invalid-title}")
+	propBadTitle := createNewProposal(t, id, []www.File{*mdBadTitle})
+
+	// Too many index files
+	propMaxIndexFiles := createNewProposal(t, id, []www.File{*md, *md})
+
+	// Index file too large
+	indexFileLarge := createFileMD(t, www.PolicyMaxIndexFileSize, "Valid Title")
+	propIndexLarge := createNewProposal(t, id, []www.File{*indexFileLarge, *png})
+
+	// Too many attached files. We need one correctly named md
 	// file and the rest must have their names changed so that we
 	// don't get a duplicate filename error.
-	files := make([]www.File, 0, www.PolicyMaxMDs+1)
+	files := make([]www.File, 0, www.PolicyMaxAttachments+1)
 	files = append(files, *md)
-	for i := 0; i < www.PolicyMaxMDs; i++ {
+	for i := 0; i < www.PolicyMaxAttachments+1; i++ {
 		m := *md
 		m.Name = fmt.Sprintf("%v.md", i)
 		files = append(files, m)
 	}
-	propMaxMDFiles := createNewProposal(t, id, files)
-
-	// Too many image files. All of their names must be different
-	// so that we don't get a duplicate filename error.
-	files = make([]www.File, 0, www.PolicyMaxImages+2)
-	files = append(files, *md)
-	for i := 0; i <= www.PolicyMaxImages; i++ {
-		p := *png
-		p.Name = fmt.Sprintf("%v.png", i)
-		files = append(files, p)
-	}
-	propMaxImages := createNewProposal(t, id, files)
-
-	// Markdown file too large
-	mdLarge := createFileMD(t, www.PolicyMaxMDSize, "Valid Title")
-	propMDLarge := createNewProposal(t, id, []www.File{*mdLarge, *png})
+	propMaxAttachments := createNewProposal(t, id, files)
 
-	// Image too large
-	pngLarge := createFilePNG(t, true)
-	propImageLarge := createNewProposal(t, id, []www.File{*md, *pngLarge})
-
-	// Invalid proposal title
-	mdBadTitle := createFileMD(t, 8, "{invalid-title}")
-	propBadTitle := createNewProposal(t, id, []www.File{*mdBadTitle})
+	// Attached file too large
+	fileLarge := createFilePNG(t, true)
+	propAttachmentLarge := createNewProposal(t, id, []www.File{*md, *fileLarge})
 
 	// Setup test cases
 	var tests = []struct {
```

Comment:
`TestValidateProposal` is missing the test cases:

"invalid public key"
"no index file"
"invalid index filename"
"invalid index mime type"

Note: "no files" is not the same as "no index file".  They both need to be tested.

Reply:
Pushing the updated tests now. Had to update validation methods in order to add "invalid index mime type" test, as the index file weren't being checked for their mime type. Refactored overall code for invoices_test, much better now :+1: